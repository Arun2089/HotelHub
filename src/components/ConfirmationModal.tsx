import { Button } from "@mui/material";

import DialogModal from "./DialogModal";
import LoadingButton from "./LoadingButton";

type DialogModalPropsTypes = {
  isOpen: boolean;
  title: string;
  buttontext: string;
  buttonType: string;
  children: React.ReactNode;
  handleClose: () => void;
  handleSubmit: () => void;
};

const ConfirmationModal = ({
  isOpen,
  title,
  buttontext,
  buttonType,
  children,
  handleClose,
  handleSubmit,
}: DialogModalPropsTypes) => {
  const actions: React.ReactNode = (
    <>
      <Button variant="outlined" size="large" onClick={handleClose}>
        Cancel
      </Button>
      <LoadingButton
        buttonText={buttontext}
        onSubmit={handleSubmit}
        color={buttonType === "delete" ? "error" : "primary"}
      />
    </>
  );

  return (
    <DialogModal
      dialogSize="xs"
      isOpen={isOpen}
      title={title}
      handleClose={handleClose}
      actions={actions}
    >
      {children}
    </DialogModal>
  );
};

export default ConfirmationModal;
