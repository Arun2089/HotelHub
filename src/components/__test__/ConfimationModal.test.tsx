import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import ConfirmationModal from "../ConfirmationModal";

const handleClose = jest.fn();
const handleSubmit = jest.fn();

describe("ConfirmationModal", () => {
  it("snapshot test", () => {
    const { asFragment } = render(
      <ConfirmationModal
        isOpen={true}
        title="Test Title"
        buttontext="Confirm"
        buttonType="default"
        handleClose={handleClose}
        handleSubmit={handleSubmit}
      >
        <p>Test content</p>
      </ConfirmationModal>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    render(
      <ConfirmationModal
        isOpen={true}
        title="Test Title"
        buttontext="Confirm"
        buttonType="default"
        handleClose={handleClose}
        handleSubmit={handleSubmit}
      >
        <p>Test content</p>
      </ConfirmationModal>
    );

    expect(screen.getByText("Test Title")).toBeInTheDocument();
    expect(screen.getByText("Confirm")).toBeInTheDocument();
    expect(screen.getByText("Cancel")).toBeInTheDocument();
    expect(screen.getByText("Test content")).toBeInTheDocument();
  });

  it("calls handleClose on cancel button or cross icon click clicks", async () => {
    render(
      <ConfirmationModal
        isOpen={true}
        title="Test Title"
        buttontext="Confirm"
        buttonType="default"
        handleClose={handleClose}
        handleSubmit={handleSubmit}
      >
        <p>Test content</p>
      </ConfirmationModal>
    );

    fireEvent.click(screen.getByText("Cancel"));
    fireEvent.click(screen.getByTestId("close-icon"));

    expect(handleClose).toHaveBeenCalledTimes(2);
  });

  it("calls handleSubmit on submit button clicks", async () => {
    render(
      <ConfirmationModal
        isOpen={true}
        title="Test Title"
        buttontext="Confirm"
        buttonType="default"
        handleClose={handleClose}
        handleSubmit={handleSubmit}
      >
        <p>Test content</p>
      </ConfirmationModal>
    );

    fireEvent.click(screen.getByText("Confirm"));
    await waitFor(() => expect(handleSubmit).toHaveBeenCalledTimes(1));
  });
});
