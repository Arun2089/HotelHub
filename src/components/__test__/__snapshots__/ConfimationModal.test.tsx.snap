// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal snapshot test 1`] = `<DocumentFragment />`;
